### ISSUE TITLE
How to replicate the performance of transformer

### ISSUE THREAD

**Issue by gaopengcuhk** [pytorch/fairseq#1352]
srun --pty --gres gpu:8 -c 16 fairseq-train     data-bin/wmt16_en_de_bpe32k     --arch transformer --share-all-embeddings     --optimizer adam --adam-betas '(0.9, 0.98)' --clip-norm 0.0     --lr 0.0006 --lr-scheduler inverse_sqrt --warmup-updates 4000 --warmup-init-lr 1e-07     --dropout 0.1 --weight-decay 0.0     --criterion label_smoothed_cross_entropy --label-smoothing 0.1     --max-tokens 5500 --fp16

I want to replicate the base transformer performance on WMT EN-DE

However, I can only get 26.65 BLUE score.

Can you give me any suggestion?

**Comment by myleott** [comment]
The original Attention Is All You Need paper reported 27.3 BLEU, but that included several tricks:
- checkpoint averaging: you should try averaging the last few checkpoints using the average_checkpoints script: https://github.com/pytorch/fairseq/blob/master/scripts/average_checkpoints.py
- during generation make sure you're using `--beam 4 --lenpen 0.6 --remove-bpe`
- compound splitting: you will get slightly higher BLEU scores by using compound splitting. Save the output from generate.py and then use the compound_split_bleu.sh script: https://github.com/pytorch/fairseq/blob/master/scripts/compound_split_bleu.sh

**Comment by huihuifan** [comment]
closing this issue after myle's answer
